### PR TITLE
fix(zimwriterfs): manage magic lifecycle via MagicInitializer helper

### PR DESCRIPTION
**Pull request by lakshyajn**
Extracted from #538 as requested by @veloman-yunkan to isolate `magic`-related fixes into a dedicated PR before transitioning to `std::filesystem`.

### Changes
- **RAII `MagicInitializer`:** Added a file-scope static `MagicInitializer` in `src/zimwriterfs/tools.cpp` to automatically open/load `magic` on startup and close it on exit.
- **Deduplication:** Removed manual `magic_open` / `magic_load` / `magic_close` boilerplate from `src/zimwriterfs/zimwriterfs.cpp` and unit tests.
- **Null Safety:** Added guards against a `nullptr` `magic` handle and null return value from `magic_file()` in `getMimeTypeForFile()`.
- **Unit Tests:** Added unit test coverage for `getMimeTypeForFile()` in `test/tools-test.cpp` without needing manual setup/teardown boilerplate.

### CODE REVIEW

**Comment by lakshyajn** [comment]
Keeping this minimal for now to extract the magic lifecycle changes from #538 as requested by @veloman-yunkan  without adding new test fixtures.

File: test/zimwriterfs-zimcreatorfs.cpp [hunk 1]
```diff
@@ -33,24 +33,6 @@ bool inflateHtmlFlag = false;
 bool isVerbose() { return false; }
 magic_t magic;
```
Comment (veloman-yunkan) [inline test/zimwriterfs-zimcreatorfs.cpp]: Shouldn't this be removed too?

File: test/tools-test.cpp [hunk 2]
```diff
@@ -13,6 +14,12 @@ TEST(CommonTools, isDirectory)
   EXPECT_TRUE(isDirectory("data/minimal-content"));
 }
 
+TEST(CommonTools, getMimeTypeForFile)
+{
+  // magic is initialized by the static MagicInitializer in zimwriterfs/tools.cpp
```
Comment (veloman-yunkan) [inline test/tools-test.cpp]: No need for this comment.

File: src/zimwriterfs/tools.cpp [hunk 3]
```diff
@@ -77,6 +77,31 @@ extern bool inflateHtmlFlag;
...
+        magic_close(magic);
+        magic = nullptr;
+      }
+    }
+  }
+  ~MagicInitializer() {
+    if (magic) {
+      magic_close(magic);
+      magic = nullptr;
+    }
```
Comment (veloman-yunkan) [inline src/zimwriterfs/tools.cpp:96]: Package this into a method (something like `uninit()`) and reuse in the ctor

File: src/zimwriterfs/zimwriterfs.cpp [hunk 4]
```diff
@@ -516,7 +515,7 @@ int main(int argc, char** argv)
     exit(1);
   }
 
-  magic_close(magic);
+  /* magic is closed by MagicInitializer in tools.cpp */
```
Comment (veloman-yunkan) [inline src/zimwriterfs/zimwriterfs.cpp]: No need for this comment

File: src/zimwriterfs/zimwriterfs.cpp [hunk 5]
```diff
@@ -492,9 +492,8 @@ int main(int argc, char** argv)
 {
   parse_args(argc, argv);
 
-  /* Init */
-  magic = magic_open(MAGIC_MIME);
-  if (magic_load(magic, NULL) != 0) {
+  /* Init - magic is initialized by MagicInitializer in tools.cpp */
```
Comment (veloman-yunkan) [inline src/zimwriterfs/zimwriterfs.cpp]: The need for this comment points to a defect of the current solution with automatic initialization of `magic`. After looking more closely, I see that `magic` is defined in this file (and that's why it had to be defined in the unit test file too). The definition of `magic` must be moved to `src/zimwriterfs/tools.cpp`. Then it makes sense to hide it as a static variable in a free function `magic_t& getMagic()` (declared in `src/zimwriterfs/tools.h`) that returns an initialized object via RAII.

File: src/zimwriterfs/tools.cpp [hunk 6]
```diff
@@ -225,12 +250,17 @@ std::string getMimeTypeForFile(const std::string &directoryPath, const std::stri
...
+      const std::string path = directoryPath + "/" + filename;
+      const char* magic_res = magic_file(magic, path.c_str());
+      if (magic_res) {
+        mimeType = std::string(magic_res);
+        if (mimeType.find(";") != std::string::npos) {
+          mimeType = mimeType.substr(0, mimeType.find(";"));
+        }
+        fileMimeTypes[filename] = mimeType;
+      }
     }
```
Comment (veloman-yunkan) [inline src/zimwriterfs/tools.cpp:265]: Make this change in a separate commit.

File: test/tools-test.cpp [hunk 7]
```diff
@@ -13,6 +14,12 @@ TEST(CommonTools, isDirectory)
...
+  EXPECT_EQ(getMimeTypeForFile("data/minimal-content", "favicon.png"), "image/png");
+}
+
```
Comment (veloman-yunkan) [inline test/tools-test.cpp:28]: Introduce this unit-test in a separate commit. Also cover cases requiring usage of magic (see Copilot's review).

**Review by veloman-yunkan — CHANGES_REQUESTED** [review]
Please structure the PR as follows (3 commits):

1. Improvement of initialization of magic (Introduce `magic_t& getMagic()` encapsulating access to the initialized `magic` object).
2. Add a unit test for `getMimeTypeForFile()`
3. Address edge-cases of the usage of magic in `getMimeTypeForFile()`

**Comment by kelson42** [comment]
Are we ready for a new review here?
